rollup-node.md

```diff
@@ -0,0 +1,296 @@
+entries.
+
+[deposit-feed-spec]: TODO
+
+> **TODO LINK** deposit feed contract specification
+
+From these L2 derivation inputs, the rollup node constructs an expanded version
+of the [Engine API PayloadAttributesV1 object][PayloadAttributesV1]:
+
+[PayloadAttributesV1]: https://github.com/ethereum/execution-apis/blob/main/src/engine/specification.md#payloadattributesv1
+
+```
+PayloadAttributesOPV1: {
+	timestamp: QUANTITY
+	random: DATA (32 bytes)
+	suggestedFeeRecipient: DATA (20 bytes)
+	transactions: array of DATA
+}
+```
+
+The type notation used here refers to the [HEX value encoding] used by the
+[Ethereum JSON-RPC API specification][JSON-RPC-API], as this structure will need to
+be sent over JSON-RPC. `array` refers to a JSON array.
+
+[HEX value encoding]: https://eth.wiki/json-rpc/API#hex-value-encoding
+[JSON-RPC-API]: https://eth.wiki/json-rpc/API
+
+The object properties must be set as follows:
+
+- `timestamp` is set to the current [unix time] (number of elapsed seconds since
+  00:00:00 UTC on 1 January 1970), rounded to the closest multiple of 2 seconds.
+- `random` is set to the *random* L1 block attribute
+- `suggestedFeeRecipient` is set to an address where the sequencer would like to
+  direct the fees
+- `transactions` is an array of transactions, encoded in the [EIP-2718] format
+  (i.e. as a single byte defining the transaction type, concatenated with an
+  opaque byte array whose meaning depends on the type).
+
+> **TODO** we need to handle non-EIP-2718 transactions too
+
+[unix type]: https://en.wikipedia.org/wiki/Unix_time
+[merge]: https://ethereum.org/en/eth2/merge/
+[EIP-2718]: https://eips.ethereum.org/EIPS/eip-2718
+
+[encode-tx]: https://github.com/norswap/nanoeth/blob/cc5d94a349c90627024f3cd629a2d830008fec72/src/com/norswap/nanoeth/transactions/Transaction.java#L84-L130
+
+The [EIP-2718] transactions must have a transaction type that is valid on L1, or
+be an *[L1 attributes transaction]* (see below).
+
+#### Payload Transaction Format
+[payload-format]: #payload-transaction-format
+
+The `transactions` array is filled with the deposits, prefixed by the (single)
+[L1 attributes transaction]. The deposits are simply copied byte-for-byte — it
+is the role of the [execution engine] to reject invalidly-formatted
+transactions.
+
+> **TODO** must offer some precisions on the format of deposits: sender,
+> receivers both in-tx-as-encoded, and on-L2-tx. What about the fees?
+
+The Optimistic Ethereum specific *[L1 attributes transaction]* has the following
+[EIP-2718]-compatible format: `0x7E || [block_number, timestamp, basefee]`
+where:
+
+- `0x7E` is the transaction type identifier. It is selected because transaction
+  type identifiers are currently allowed to go up to `0x7F`. Picking a high
+  identifier minimizes the risk that the identifier will be used by Ethereum in
+  the future. We don't pick `0x7F` itself in case it becomes used for a
+  variable-length encoding scheme.
```

Comment:
Is this referring to some discussion elsewhere about using 0x7F? 
Perhaps we should choose some address space closer to the middle of the range, which is a less likely schelling point than the beginning or end.

Reply:
No this is all me.

idk, we should do what seems likely. I think EIPs are likely to just pick in sequential orders.

But in the very unlikely case where they'd need a variable-length bytecode encoding, the traditional way to do that is to use the last byte in the range as the variable-length marker.

In any case, whatever we pick will be relevant to the design of future EIPs anyway, so I'm not too afraid, but I feel like this is both a bit safer & cleaner.

Reply:
Marking this resolved, feel free to open an issue if you think it warrants further discussion.